### PR TITLE
Bug 1223419 - Allow setting file permission during upload

### PR DESCRIPTION
**Pull request by eliperelman**

### CODE REVIEW

**Comment by gaye** [comment]
Ping me when you pull in latest stuff :+1: !

**Comment by gaye** [comment]
Didn't mean to hit close :D

**Comment by eliperelman** [comment]
@gaye rebased.

File: test/commands/shell.js [hunk 1]
```diff
@@ -71,6 +71,15 @@ commands.getprop = function getprop(name) {
   });
 };
 
+commands.chmod = function chmod(mode, file) {
+  mode = parseInt(mode, 10) || mode;
```
Comment (gaye) [inline test/commands/shell.js]: Can you explain this to me? (The or part mostly)

Comment (gaye) [inline test/commands/shell.js]: Oh `NaN`

Comment (gaye) [inline test/commands/shell.js]: ehhh

Comment (gaye) [inline test/commands/shell.js]: How about `mode = /^\d+$/.test(mode) ? parseInt(mode) : mode;`

Comment (eliperelman) [inline test/commands/shell.js]: Come to think of it, if we are just testing for digits, we don't need to parse to numbers at all. I'll update.

**Pull request merged**